Add reset template button to Woo email customizer [MAILPOET-6213]

costasovo opened the pull request:
## Description

This PR adds possibility to reset the WooCommerce email template to the initial state.

## Code review notes
Instead of a form or API that would reset the template on the backend, I decided to use the editor application. The advantage is that the reset is done as a single edit step and can be reverted via undo arrow.

## QA notes

1) Go to Woo Transactional Email customizer
2) Make some edits
3) Click Reset template button and the template should reset to the initial state when the plugin was installed

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-6213]

## After-merge notes

_N/A_

## Tasks

- [] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [] I added sufficient test coverage
- [] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes


[MAILPOET-6213]: https://mailpoet.atlassian.net/browse/MAILPOET-6213?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:reset-woo-template)

_The latest successful build from `reset-woo-template` will be used. If none is available, the link won't work._

costasovo committed:
Make the original Woo Transactional email template available to JS app

[MAILPOET-6213]

costasovo committed:
Unify property type annotations in Newsletter Editor page controller

[MAILPOET-6213]

costasovo committed:
Add reset template button and reset template functionality

[MAILPOET-6213]

costasovo committed:
Fix typo in the method name in DataInconsistencyRepository

[MAILPOET-6213]

lysyjan reviewed (dismissed):
Good job 👍

Aschepikov commented:
Hi @costasovo  When I make some changes with the content blocks I can't reset them, as nothing changes. To reset the new content blocks from the template I need to reload the page and then the reset button will renew it. Is it expected? 
![image](https://github.com/user-attachments/assets/5f994b6b-9d2d-45a2-94e9-e44d6811c514)

costasovo commented:
> When I make some changes with the content blocks I can't reset them, as nothing changes. To reset the new content blocks from the template I need to reload the page and then the reset button will renew it. Is it expected?

@Aschepikov You should always be able to reset to the original template. Could you please provide steps how to replicate the problem?

Aschepikov commented:
@costasovo  1. To change something in the styles 
2. To reset the template (it will work)
3. To change something in the content or the columns
4. To reset the template (now it wont work)

costasovo committed:
Fix reset template functionality to work for multiple reset

The issue was that after the first reset the global template object
became a model and all changes were save also to this object so
subsequent resets didn't work.

This commit fixes it by making sure we clone the original template before resetting.
[MAILPOET-6213]

Aschepikov commented:
@costasovo  I rechecked it and now it is working perfectly, but I see a tiny issue the reset template button changes the template to the completed order but not changes the name in the dropdown. Should we fix it?
Steps to reproduce:
1. Open the new account template 
![image](https://github.com/user-attachments/assets/469f2bec-4d06-4190-b936-b361ba4a2cb0)

2. Change something and reset the template
3. Now it changes the template to the completed order but in the dropdown, you can see the new account template name
![image](https://github.com/user-attachments/assets/e8d234ee-cfe8-49a8-8c4a-19690f01b256)

costasovo committed:
Preserve selected Woo email dummy content type on history undo/redo

[MAILPOET-6213]

Aschepikov reviewed (approved):
GJ!
